sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -627,8 +644,32 @@ && getConnectionInitSqls().get() != null
         if (getMaxConnections() != null && getMaxConnections().get() != null) {
           basicDataSource.setMaxTotal(getMaxConnections().get());
         }
-        if (getDriverClassLoader() != null) {
-          basicDataSource.setDriverClassLoader(getDriverClassLoader());
+        if (getDriverJarDirectory() != null && getDriverJarDirectory().get() != null) {
+          try {
+            String pathPrefix =
+                getDriverJarDirectory().get().endsWith("/")
+                    ? getDriverJarDirectory().get()
+                    : getDriverJarDirectory().get() + "/";
+            MatchResult result = FileSystems.match(pathPrefix + "**", EmptyMatchTreatment.ALLOW);
```

Comment:
I'm not sure if the FileSystems API will work correctly at job submission without initialization (it's intended to be primarily used at workers). Did we test using GCS etc. ?